oneview_redfish_toolkit/blueprints/chassis.py

```diff
@@ -44,55 +47,92 @@ def get_chassis(uuid):
+            raise OneViewRedfishError(
+                {"errorCode": "INVALID_INFORMATION",
+                 "message": "Invalid JSON key: {}".format(invalid_key)})
+
         resource_index = g.oneview_client.index_resources.get_all(
             filter='uuid=' + uuid
         )
 
-        if resource_index:
-            category = resource_index[0]["category"]
-        else:
-            raise OneViewRedfishError('Cannot find Index resource')
-
+        category = resource_index[0]["category"]
         if category == 'server-hardware':
-            server_hardware = g.oneview_client.server_hardware.get(uuid)
-            etag = server_hardware['eTag']
-            ch = BladeChassis(server_hardware)
-        elif category == 'enclosures':
-            enclosure = g.oneview_client.enclosures.get(uuid)
-            etag = enclosure['eTag']
-            enclosure_environment_config = g.oneview_client.enclosures.\
-                get_environmental_configuration(uuid)
-            ch = EnclosureChassis(
-                enclosure,
-                enclosure_environment_config
-            )
-        elif category == 'racks':
-            racks = g.oneview_client.racks.get(uuid)
-            etag = racks['eTag']
-            ch = RackChassis(racks)
-        else:
-            raise OneViewRedfishError('Chassis type not found')
+            # Gets ServerHardware for given UUID
+            sh = g.oneview_client.server_hardware.get(uuid)
 
-        json_str = ch.serialize()
+            # Gets power configuration based on OneView pattern
+            oneview_power_configuration = \
+                OneViewPowerOption.get_oneview_power_configuration(
+                    sh, reset_type
+                )
 
-        response = Response(
-            response=json_str,
-            status=status.HTTP_200_OK,
-            mimetype="application/json")
-        response.headers.add("ETag", "W/" + etag)
-        return response
-    except HPOneViewException as e:
-        # In case of error log exception and abort
-        logging.exception(e)
-        abort(status.HTTP_404_NOT_FOUND)
+            # Changes the ServerHardware power state
+            g.oneview_client.server_hardware.update_power_state(
+                oneview_power_configuration, uuid)
+
+            return Response(
+                response='{"ResetType": "%s"}' % reset_type,
+                status=status.HTTP_200_OK,
+                mimetype='application/json')
 
     except OneViewRedfishError as e:
         # In case of error log exception and abort
-        logging.exception('Unexpected error: {}'.format(e))
-        abort(status.HTTP_404_NOT_FOUND, "Chassis not found")
+        logging.exception('Mapping error: {}'.format(e))
 
-    except Exception as e:
-        # In case of error log exception and abort
-        logging.exception('Unexpected error: {}'.format(e))
-        abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
+        if e.msg["errorCode"] == "NOT_IMPLEMENTED":
```

Comment:
@ricardogpsf maybe we could place this verification in the ResponseBuilder or as a new flask app handler.

Reply:
Yes. I agree with you, but we need a more time to think how to change this, so I created this issue to be done later: https://github.com/HewlettPackard/oneview-redfish-toolkit/issues/342